covid19_sfbayarea/data/utils.py

```diff
@@ -48,8 +53,37 @@ def request(self, url: str, params: Dict = None, **kwargs: Any) -> Dict:
 
         return self._request(url, **kwargs)
 
-    def resource(self, resource_id: str, **kwargs: Any) -> Dict:
-        return self.request(f'{self.resource_url}{resource_id}', **kwargs)
+    def resource(self, resource_id: str, **kwargs: Any) -> List[Dict]:
+        data: List[Dict] = []
+
+        if "params" in kwargs:
+            if "$offset" not in kwargs["params"]:
+                kwargs["params"].update({"$offset": 0})
+
+            else:
+                pass
+
+        else:
+            kwargs["params"] = {"$offset": 0}
```

Comment:
I think you could simplify the logic here if you pulled `params` out off `kwargs`. i.e. make the method signature:

```py
def resource(self, resource_id: str, params: dict = None, **kwargs: Any) -> List[Dict]:
```

And then the logic here can be:

```py
    params = params or {}
    params.setdefault("$offset", 0)
    params.setdefault("$limit", self.DEFAULT_LIMIT)

    # ...and down where you make the request:
    results = self.request(f'{self.resource_url}{resource_id}', params=params, **kwargs)
```

As a side note: I think you should explicitly set the `$limit` query param here; we shouldn’t rely on all Socrata servers using the same default or that future updates will keep this default.